qiskit/tools/visualization/bloch.py

```diff
@@ -0,0 +1,652 @@
+    """Makes a fancy arrow"""
+    def __init__(self, xs, ys, zs, *args, **kwargs):
+        FancyArrowPatch.__init__(self, (0, 0), (0, 0), *args, **kwargs)
+        self._verts3d = xs, ys, zs
+
+    def draw(self, renderer):
+        xs3d, ys3d, zs3d = self._verts3d
+        xs, ys, _ = proj3d.proj_transform(xs3d, ys3d, zs3d, renderer.M)
+        self.set_positions((xs[0], ys[0]), (xs[1], ys[1]))
+        FancyArrowPatch.draw(self, renderer)
+
+
+class Bloch():
+    """Class for plotting data on the Bloch sphere.  Valid data can be
+    either points, vectors, or qobj objects.
+    Attributes
+    ----------
+    axes : instance {None}
+        User supplied Matplotlib axes for Bloch sphere animation.
+    fig : instance {None}
+        User supplied Matplotlib Figure instance for plotting Bloch sphere.
+    font_color : str {'black'}
+        Color of font used for Bloch sphere labels.
+    font_size : int {20}
+        Size of font used for Bloch sphere labels.
+    frame_alpha : float {0.1}
+        Sets transparency of Bloch sphere frame.
+    frame_color : str {'gray'}
+        Color of sphere wireframe.
+    frame_width : int {1}
+        Width of wireframe.
+    point_color : list {["b","r","g","#CC6600"]}
+        List of colors for Bloch sphere point markers to cycle through.
+        i.e. By default, points 0 and 4 will both be blue ('b').
+    point_marker : list {["o","s","d","^"]}
+        List of point marker shapes to cycle through.
+    point_size : list {[25,32,35,45]}
+        List of point marker sizes. Note, not all point markers look
+        the same size when plotted!
+    sphere_alpha : float {0.2}
+        Transparency of Bloch sphere itself.
+    sphere_color : str {'#FFDDDD'}
+        Color of Bloch sphere.
+    figsize : list {[7,7]}
+        Figure size of Bloch sphere plot.  Best to have both numbers the same;
+        otherwise you will have a Bloch sphere that looks like a football.
+    vector_color : list {["g","#CC6600","b","r"]}
+        List of vector colors to cycle through.
+    vector_width : int {5}
+        Width of displayed vectors.
+    vector_style : str {'-|>', 'simple', 'fancy', ''}
+        Vector arrowhead style (from matplotlib's arrow style).
+    vector_mutation : int {20}
+        Width of vectors arrowhead.
+    view : list {[-60,30]}
+        Azimuthal and Elevation viewing angles.
+    xlabel : list {["$x$",""]}
+        List of strings corresponding to +x and -x axes labels, respectively.
+    xlpos : list {[1.1,-1.1]}
+        Positions of +x and -x labels respectively.
+    ylabel : list {["$y$",""]}
+        List of strings corresponding to +y and -y axes labels, respectively.
+    ylpos : list {[1.2,-1.2]}
+        Positions of +y and -y labels respectively.
+    zlabel : list {[r'$\\left|1\\right>$',r'$\\left|0\\right>$']}
+        List of strings corresponding to +z and -z axes labels, respectively.
+    zlpos : list {[1.2,-1.2]}
+        Positions of +z and -z labels respectively.
```

Comment:
Can you revise the docstrings included in this PR to make sure they conform to [google-style docstrings](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html), in order to have them matched with the rest of the codebase?